lib/Cache.js

```diff
@@ -125,25 +128,27 @@ class Cache {
 				if (typeof value === 'undefined') { continue; }
 
 				for (const prop in slice.filter) {
+
 					if (
 						!this._checkFilter(entity, prop, slice.filter[prop])
 					) {
 						continue keyloop; //eslint-disable-line
 					}
 				}
-				if (this.knowledge[key].contains(value)) {
-					keyValues[key] = { value, item: entity };
-				}
 
-				if (slice.link) { for (const linkType in slice.link) {
-					const dKey = this.getDualKey(slice, linkType);
+				keyValues[key] = { value, item: entity };
```

Comment:
It is being handled in the putEntity function.
We delete the original item and add the new item.
The knowledge generated by the result of changing order property from a value in range to something out of range has two ranges.

Example: 
this.knowledge for a key is say [ [1, 6] ] and I change the order property from 3 to 7.

allChanges now has the a range array that looks like [ [3,3], [7,7] ].